packages/react-native/scripts/cocoapods/privacy_manifest_utils.rb

```diff
@@ -0,0 +1,159 @@
+            reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"]
+            required_reason_apis[api_type] ||= []
+            required_reason_apis[api_type] += reasons
+        end
+
+        # Merge the Required Reason APIs from pods with the ones from the existing PrivacyInfo file
+        (privacy_info["NSPrivacyAccessedAPITypes"] || []).each do |accessed_api|
+            api_type = accessed_api["NSPrivacyAccessedAPIType"]
+            reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"]
+            # Add reasons from existing PrivacyInfo file to the ones from pods
+            required_reason_apis[api_type] ||= []
+            required_reason_apis[api_type] += reasons
+        end
+
+        # Update the existing PrivacyInfo file with the new aggregated data
+        privacy_info["NSPrivacyAccessedAPITypes"] = required_reason_apis.map { |api_type, reasons|
+            {
+            "NSPrivacyAccessedAPIType" => api_type,
+            "NSPrivacyAccessedAPITypeReasons" => reasons.uniq
+            }
+        }
+
+        Xcodeproj::Plist.write_to_path(privacy_info, file_path)
+
+        ensure_reference(file_path, user_project, target)
+    end
+
+    def self.get_application_target(user_project)
+        return user_project.targets.first{ |t| t.symbol_type == :application }    
+    end
+
+    def self.read_privacyinfo_file(file_path)
+        # Maybe add missing default NSPrivacyTracking, NSPrivacyTrackingDomains, NSPrivacyCollectedDataTypes, but this works without those keys
+        source_data = nil
+        # Try to read an existing PrivacyInfo.xcprivacy file
+        begin
+            source_data = Xcodeproj::Plist.read_from_path(file_path)
+            Pod::UI.puts "[Privacy Manifest Aggregation] Appending aggregated reasons to existing PrivacyInfo.xcprivacy file."
+        rescue => e
+            Pod::UI.puts "[Privacy Manifest Aggregation] No existing PrivacyInfo.xcprivacy file found, creating a new one."
+        end
+        return source_data
+    end
+
+    def self.ensure_reference(file_path, user_project, target)
+        reference_exists = target.resources_build_phase.files_references.any? { |file_ref| file_ref.path.end_with? "PrivacyInfo.xcprivacy" }
+        unless reference_exists
+            # We try to find the main group, but if it doesn't exist, we default to adding the file to the project root – both work
+            file_root = user_project.root_object.main_group.children.first { |group| group.name == target.name } || user_project
+            file_ref = file_root.new_file(file_path)
+            build_file = target.resources_build_phase.add_file_reference(file_ref, true)
+        end
+    end
+
+    def self.get_privacyinfo_file_path(target, user_project)
+        # We try to find a file we know exists in the project to get the path to the main group directory
+        app_delegate_path = target.source_build_phase.files_references.find { |file_ref| file_ref.name == "Info.plist" }
```

Comment:
```suggestion
        info_plist_path = target.source_build_phase.files_references.find { |file_ref| file_ref.name == "Info.plist" }
```

Reply:
I also changed it to read it from the user_project files, since info.plist does not end up in `source_build_phase` after all.